File: deck/seed/code/native/kotlin/runtime/directory.kt
// Working directory runtime. Reached only through the public environment API.
//
// The JVM has no real chdir: `user.dir` is read at startup and setting it only affects code that resolves relative
// paths through it. That limitation is the platform's, and it is kept here rather than surfacing in the seed source.
object directory {
    fun get(): String = System.getProperty("user.dir") ?: ""

    fun set(path: String) {
        System.setProperty("user.dir", path)
    }
}

File: deck/seed/code/native/rust/runtime/directory.rs
// Working directory runtime. Reached only through the public environment API.
mod directory {
    pub fn get() -> String {
        std::env::current_dir()
            .map(|p| p.to_string_lossy().to_string())
            .unwrap_or_default()
    }

    pub fn set(path: String) {
        let _ = std::env::set_current_dir(&path);
    }
}
